feat(medication): EPI-1093: Add medication administration schedule to prescription form

khoi-bes opened the pull request:
### Changes

_Add a brief description of the changes in this PR to help give the reviewer context._

### Deploys

- [ ] **Deploy to Tamanu Internal** <!-- #deploy -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

khoi-bes committed:
feat(medication): EPI-1099: Updates to Frequency logic and configuration

khoi-bes committed:
Merge branch 'epic-medication-enhancements-and-mar' into feature/epi-1099-updates-to-frequency-logic-and-configuration

khoi-bes committed:
feat(prescriptions): Add ideal_times field and update related documentation

huytranbes commented on packages/constants/src/medications.ts:
```diff
@@ -220,3 +220,18 @@ export const MEDICATION_DURATION_UNITS_LABELS = {
+  { startTime: '16:00', endTime: '18:00' },
+  { startTime: '18:00', endTime: '20:00', mealTime: 'dinner' },
+  { startTime: '20:00', endTime: '22:00' },
+  { startTime: '22:00', endTime: '24:00', mealTime: 'night' },
```
`mealTime` isn't appropriate for 'Night'. Maybe use `periodLabel`

huytranbes commented on packages/web/app/forms/MedicationForm.jsx:
```diff
@@ -91,16 +128,300 @@ const ButtonRow = styled.div`
+  const handleResetToDefault = () => {
+    if (
+      values.frequency === ADMINISTRATION_FREQUENCIES.AS_DIRECTED ||
+      values.frequency === ADMINISTRATION_FREQUENCIES.IMMEDIATELY
```
Can you put this in a constant to avoid duplication. E.g
`export const NON_REPEATING_FREQUENCIES = [ADMINISTRATION_FREQUENCIES.AS_DIRECTED, ADMINISTRATION_FREQUENCIES.IMMEDIATELY]`

huytranbes commented on packages/web/app/forms/MedicationForm.jsx:
```diff
@@ -91,16 +128,300 @@ const ButtonRow = styled.div`
+                      borderRadius="3px"
+                      width="187px"
+                      height="fit-content"
+                      border={`1px solid ${checked ? Colors.primary : Colors.outline}`}
```
Optional: Please use styled-component for better readability

huytranbes reviewed (approved):
Did you put the wrong task? It should be https://linear.app/bes/issue/EPI-1093/add-medication-administration-schedule-to-prescription-form
Approved but please fix the above issues before merging

khoi-bes committed:
feat(medication): EPI-1098: Validation message for frequency/administration schedule discrepancy (#7270)

feat(medication): EPI-1098: Validation message for frequency/administration schedule discrepancy
